tdrs-frontend/src/actions/reports.js

```diff
@@ -1,19 +1,101 @@
 import { logErrorToServer } from '../utils/eventLogger'
 
 import { v4 as uuidv4 } from 'uuid'
+import axios from 'axios'
 
 export const SET_FILE = 'SET_FILE'
 export const CLEAR_FILE = 'CLEAR_FILE'
 export const SET_FILE_ERROR = 'SET_FILE_ERROR'
 export const CLEAR_ERROR = 'CLEAR_ERROR'
 
+export const START_FILE_DOWNLOAD = 'START_FILE_DOWNLOAD'
+export const FILE_DOWNLOAD_ERROR = 'FILE_DOWNLOAD_ERROR'
+
+export const FETCH_FILE_LIST = 'FETCH_FILE_LIST'
+export const SET_FILE_LIST = 'SET_FILE_LIST'
+export const FETCH_FILE_LIST_ERROR = 'FETCH_FILE_LIST_ERROR'
+export const DOWNLOAD_DIALOG_OPEN = 'DOWNLOAD_DIALOG_OPEN'
+
 export const clearFile = ({ section }) => (dispatch) => {
   dispatch({ type: CLEAR_FILE, payload: { section } })
 }
 
 export const clearError = ({ section }) => (dispatch) => {
   dispatch({ type: CLEAR_ERROR, payload: { section } })
 }
+/**
+   Get a list of files that can be downloaded, mainly used to decide
+   if the download button should be present.
+*/
+export const getAvailableFileList = ({ year, quarter = 'Q1' }) => async (
+  dispatch
+) => {
+  dispatch({
+    type: FETCH_FILE_LIST,
+  })
+  try {
+    const response = await axios.get(`/mock_api/reports/${year}/${quarter}`, {
+      responseType: 'json',
+    })
+    dispatch({
+      type: SET_FILE_LIST,
+      payload: {
+        data: response.data,
+      },
+    })
+  } catch (error) {
+    dispatch({
+      type: FETCH_FILE_LIST_ERROR,
+      payload: {
+        error,
+        year,
+        quarter,
+      },
+    })
+  }
+}
+
+export const download = ({ year, quarter = 'Q1', section }) => async (
+  dispatch
+) => {
+  try {
+    if (!year) throw new Error('No year was provided to download action.')
+    dispatch({ type: START_FILE_DOWNLOAD })
+
+    const response = await axios.get(
+      `/mock_api/reports/data-files/${year}/${quarter}/${section}`,
+      {
+        responseType: 'blob',
+      }
+    )
+    const data = response.data
+
+    // Create a link and associate it with the blob returned from the file
+    // download - this allows us to trigger the file download dialog without
+    // having to change the route or reload the page.
+    const url = window.URL.createObjectURL(new Blob([data]))
+    const link = document.createElement('a')
+
+    link.href = url
+    link.setAttribute('download', `${year}.${quarter}.${section}.txt`)
```

Comment:
i like that we're using the report file metadata in the filename of the downloaded file because STTs will (at least for now) probably maintain those standard naming conventions, and this makes it easer to distinguish them. A couple of thoughts for possible future enhancements:
- may be worth asking STT users in a future workshop if the download file names are recognizable or if they have any preference for including the original filename in the name
- I'd like to see to see year prefixed with "FY" (fiscal year) so its crystal clear that this is different from calendar year (something STTs have expressed confusion on), but this is another thing that would be good for STT users to react to now that the feature is here 😄 
 cc: @dk-ui